openapi.yaml

```diff
@@ -4027,6 +4019,49 @@ components:
           type: string
           description: Used as a human-readable label for a link.
           example: openEO
+    asset:
+      type: object
+      required:
+        - href
+      properties:
+        href:
+          title: Asset location
+          description: >-
+            URL to the downloadable asset.
+
+            The URLs SHOULD be available without authentication so that
+            external clients can download them easily.
+            If the data is confidential, signed URLs SHOULD be used to protect
+            against unauthorized access from third parties.
+          type: string
+        title:
+          description: The displayed title for clients and users.
+          type: string
+        description:
+          type: string
+          description: |-
+            Multi-line description to explain the asset.
+    
+            [CommonMark 0.29](http://commonmark.org/) syntax MAY be used for rich
+            text representation.
+        type:
+          title: Media Type
+          description: Media type of the asset.
+          type: string
+          example: image/tiff; application=geotiff
+        roles:
+          type: array
+          items:
+            type: string
+          description: |-
+            Purposes of the asset. Can be any value, but commonly used values are:
+            
+            * `thumbnail`: A visualization of the data, usually a lower-resolution true color image in JPEG or PNG format.
+            * `reproducibility`: Information how the data was produced and/or can be reproduced, e.g. the process graph used to compute the data in JSON format.
+            * `data`: The computed data in the format given by the user in the user-defined process.
```

Comment:
maybe I'm missing something, but how are (collection-level) assets linked to user-defined processes?

Reply:
This OpenAPI schema definition for assets is used in GET /collections/:id (collection assets) and GET /jobs/:id/results (item assets). It doesn't make much sense for collections, but it makes sense in the results where you link to the processed data. It's always hard to decide whether it makes sense to duplicate a lot of schema code just for changing the description a bit... :-|

Reply:
```suggestion
            * `data`: The computed data in the format given by the user in the user-defined process (applicable in `GET /jobs/{job_id}/results` only).
```

Reply:
I understand, but even for `/jobs`: that's not tied to user-defined processes either, right?

Reply:
Everything you pass in the body for `process` is basically a user-defined process. They don't necessarily need to be stored at /process_graphs. :-) See https://open-eo.github.io/openeo-api/draft/index.html#section/Processes

Reply:
then I would rather talk of `in the format specified in the process graph` (sounds more concrete to me) instead of using "user defined processes" (which has a more abstract ring to it)

Reply:
Although I feel like I should say `user-defined process`, similar phrases in the API also say `process graph` at the moment, so changing it to `process graph` until I may change all occurrences.